Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -455,35 +455,20 @@ - (void)_updateScreenTimeViewGeometry
     [[_screenTimeWebpageController view] setFrame:bounds];
 }
 
-- (void)_updateScreenTimeShieldVisibilityForWindow
+- (void)_updateScreenTimeBasedOnWindowVisibility
 {
     BOOL viewIsInWindow = !!self.window;
-
-    BOOL showsSystemScreenTimeBlockingView = [_configuration _showsSystemScreenTimeBlockingView];
-
-    if (viewIsInWindow) {
-        if (!showsSystemScreenTimeBlockingView && _screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show system shield.");
-        }
-    } else {
-        if (_screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide system shield.");
-        }
-    }
-
     BOOL viewIsVisible = viewIsInWindow;
+
 #if PLATFORM(MAC)
     viewIsVisible &= ((self.window.occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
 #endif
     [_screenTimeWebpageController setSuppressUsageRecording:(![_configuration websiteDataStore].isPersistent || !viewIsVisible)];
+
+    if (viewIsInWindow)
+        [self _installScreenTimeWebpageController];
```

Comment:
We should probably rename `_installScreenTimeWebpageController` to `_installScreenTimeWebpageControllerIfNeeded` now.

Reply:
Also, I think this code should be before the call to `setSuppressUsageRecording`? Otherwise you could have a `nil` `_screenTimeWebpageController`, skipping the suppression, and then it becomes non-nil after this code runs.